modules/common/networkattachment/networkattachment.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/pod"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// GetNADWithName - Get network-attachment-definition with name in namespace
+func GetNADWithName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+	namespace string,
+) (*networkv1.NetworkAttachmentDefinition, error) {
+
+	nad := &networkv1.NetworkAttachmentDefinition{}
+
+	err := h.GetClient().Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, nad)
+	if err != nil {
+		err = fmt.Errorf("Error getting network-attachment-definition %s/%s - %w", name, namespace, err)
+
+		return nil, err
+	}
+
+	return nad, nil
+}
+
+// CreateNetworksAnnotation returns pod annotation for network-attachment-definition list
+// e.g. k8s.v1.cni.cncf.io/networks: '[{"name": "internalapi", "namespace": "openstack"},{"name": "storage", "namespace": "openstack"}]'
+func CreateNetworksAnnotation(namespace string, nads []string) (map[string]string, error) {
+
+	netAnnotations := []networkv1.NetworkSelectionElement{}
+	for _, nad := range nads {
+		netAnnotations = append(
+			netAnnotations,
+			networkv1.NetworkSelectionElement{
+				Name:      nad,
+				Namespace: namespace,
+			},
+		)
+	}
+
+	networks, err := json.Marshal(netAnnotations)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode networks %s into json: %w", nads, err)
+	}
+
+	return map[string]string{networkv1.NetworkAttachmentAnnot: string(networks)}, nil
+}
+
+// GetNetworkStatusFromAnnotation returns NetworkStatus list with networking details the pods are attached to
+func GetNetworkStatusFromAnnotation(annotations map[string]string) ([]networkv1.NetworkStatus, error) {
+
+	var netStatus []networkv1.NetworkStatus
+
+	if netStatusAnnotation, ok := annotations[networkv1.NetworkStatusAnnot]; ok {
+		err := json.Unmarshal([]byte(netStatusAnnotation), &netStatus)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode networks status %s: %w", netStatusAnnotation, err)
+		}
+	}
+
+	return netStatus, nil
+}
+
+// VerifyNetworkStatusFromAnnotation - verifies if NetworkStatus annotation for the pods of a deployment,
+// pods identified via the service label selector, matches the passed in network attachments and the number of
+// per network IPs the ready count of the deployment. Return true if count matches with the list of IPs per network.
+func VerifyNetworkStatusFromAnnotation(
+	ctx context.Context,
+	helper *helper.Helper,
+	networkAttachments []string,
+	serviceLabels map[string]string,
+	readyCount int32,
+) (bool, map[string][]string, error) {
+
+	networkReady := true
+	networkAttachmentStatus := map[string][]string{}
+	if len(networkAttachments) > 0 {
+		podList, err := pod.GetPodListWithLabel(ctx, helper, helper.GetBeforeObject().GetNamespace(), serviceLabels)
+		if err != nil {
+			return networkReady, networkAttachmentStatus, err
```

Comment:
Should this maybe return false instead of the default `networkReady = true` in the case of an error?  I guess it doesn't really matter though since the caller would probably be checking the error first before even considering the boolean.

Reply:
done